pilot/pkg/model/sidecar.go

```diff
@@ -852,10 +852,27 @@ func computeWildcardHostVirtualServiceIndex(virtualServices []config.Config, ser
 	for _, vs := range virtualServices {
 		v := vs.Spec.(*networking.VirtualService)
 		for _, h := range v.Hosts {
+			// We may have duplicate (not just overlapping) hosts; in the case of exact duplicates, take the oldest one first
 			if host.Name(h).IsWildCarded() {
-				wildcardVirtualServiceHostIndex[host.Name(h)] = vs
+				existingVs, exists := wildcardVirtualServiceHostIndex[host.Name(h)]
+				if !exists {
+					wildcardVirtualServiceHostIndex[host.Name(h)] = vs
+					continue
+				}
+				// Only replace if this VS is older than the existing one for this hostname
+				if vs.GetCreationTimestamp().Before(existingVs.GetCreationTimestamp()) {
```

Comment:
As the virtualServices are ordered by two dimensions exportTo and creation timestamps. I am thinking this could introduce a regression it could select a private namespace vs before, but with the patch, it will select the older vs.

Reply:
I think we're good because the VirtualServices are prefiltered here: https://github.com/istio/istio/blob/8c74063ea7ab759339a5697be01c03afdce3a2c5/pilot/pkg/model/sidecar.go#L220

Reply:
Yeah,  i think you mean `	defaultEgressListener.virtualServices = ps.VirtualServicesForGateway(configNamespace, constants.IstioMeshGateway)
`
but if you look into the implementation it did sort by private to proxy namespace, explicitly exportto the proxy namespace and public dimensions

Reply:
Hm I think I see what you mean. So this patch/teat isn't sufficient; there needs to be another export to dimension